### PR TITLE
fix: OOM in bulkCreate with bigint value (v6)

### LINKED ISSUE #15053: Bulk create with providing bigint identifier to autoincrement column leads to OOM
## Issue Creation Checklist

- [x] I understand that my issue will be automatically closed if I don't fill in the requested information
- [x] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

## Bug Description

formatResults method in MySQL has such piece of code:

```javascript
          for (let i = startId; i < startId + data.affectedRows; i++) {
            result.push({ [this.model.rawAttributes[this.model.primaryKeyAttribute].field]: i });
          }
```

if I insert a row with value > Number.MAX_SAFE_INTEGER magic happens:
1) startId is a string, startId + data.affectedRows becomes a concatenation instead of addition
2) for id=3415718944570971483 we have a loop with 34157189445709714830 iterations
2) as we push data in array, every iteration of this infinite loop leaks memory, process crashes with OOM

### Reproducible Example

Here is the link to the SSCCE for this issue:

https://github.com/gugu/sequelize-sscce

### What do you expect to happen?

Expect bulkCreate records and return value

### What is actually happening?
```
----------------------------------------------------------
===== Running SSCCE for MYSQL with Sequelize v6.23.1 =====
----------------------------------------------------------

[Sequelize] Executed (default): DROP TABLE IF EXISTS `Foos`; [Elapsed time: 6 ms]

[Sequelize] Executed (default): SELECT CONSTRAINT_NAME as constraint_name,CONSTRAINT_NAME as constraintName,CONSTRAINT_SCHEMA as constraintSchema,CONSTRAINT_SCHEMA as constraintCatalog,TABLE_NAME as tableName,TABLE_SCHEMA as tableSchema,TABLE_SCHEMA as tableCatalog,COLUMN_NAME as columnName,REFERENCED_TABLE_SCHEMA as referencedTableSchema,REFERENCED_TABLE_SCHEMA as referencedTableCatalog,REFERENCED_TABLE_NAME as referencedTableName,REFERENCED_COLUMN_NAME as referencedColumnName FROM INFORMATION_SCHEMA.KEY_COLUMN_USAGE where TABLE_NAME = 'Foos' AND CONSTRAINT_NAME!='PRIMARY' AND CONSTRAINT_SCHEMA='sequelize_test' AND REFERENCED_TABLE_NAME IS NOT NULL; [Elapsed time: 2 ms]

[Sequelize] Executed (default): DROP TABLE IF EXISTS `Foos`; [Elapsed time: 2 ms]

[Sequelize] Executed (default): DROP TABLE IF EXISTS `Foos`; [Elapsed time: 1 ms]

[Sequelize] Executed (default): CREATE TABLE IF NOT EXISTS `Foos` (`id` BIGINT auto_increment , PRIMARY KEY (`id`)) ENGINE=InnoDB; [Elapsed time: 24 ms]

[Sequelize] Executed (default): SHOW INDEX FROM `Foos` FROM `sequelize_test` [Elapsed time: 5 ms]

[Sequelize] Executed (default): INSERT INTO `Foos` (`id`) VALUES ('3415718944570971483'),('6190095476713727669'); [Elapsed time: 8 ms]


<--- Last few GCs --->

[2853049:0x670e540]    21281 ms: Scavenge (reduce) 3956.2 (4018.7) -> 3955.2 (4018.7) MB, 3.2 / 0.0 ms  (average mu = 0.506, current mu = 0.550) allocation failure;
[2853049:0x670e540]    22979 ms: Mark-sweep (reduce) 4529.9 (4592.8) -> 4146.7 (4210.2) MB, 1415.8 / 0.0 ms  (average mu = 0.403, current mu = 0.248) allocation failure; GC in old space requested


<--- JS stacktrace --->

FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
 1: 0xb6e260 node::Abort() [node]
 2: 0xa7cdea  [node]
 3: 0xd45f00 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [node]
 4: 0xd462a7 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [node]
 5: 0xf235b5  [node]
 6: 0xf35a9d v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [node]
 7: 0xf1019e v8::internal::HeapAllocator::AllocateRawWithLightRetrySlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [node]
 8: 0xf11567 v8::internal::HeapAllocator::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [node]
 9: 0xef273a v8::internal::Factory::NewFillerObject(int, v8::internal::AllocationAlignment, v8::internal::AllocationType, v8::internal::AllocationOrigin) [node]
10: 0x12b5e0d v8::internal::Runtime_AllocateInOldGeneration(int, unsigned long*, v8::internal::Isolate*) [node]
11: 0x16e78b9  [node]
```

### Environment

- Sequelize version: 6.23.1
- Node.js version: 18.8.0
- If TypeScript related: TypeScript version:
- Database & Version: mysql 8
- Connector library & Version: mysql2 2.2.3

## Would you be willing to resolve this issue by submitting a Pull Request?

- [x] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I will need guidance.
- [ ] No, I don't have the time, but my company or I are [supporting Sequelize through donations on OpenCollective](https://opencollective.com/sequelize).
- [ ] No, I don't have the time, and I understand that I will need to wait until someone from the community or maintainers is interested in resolving my issue.

---

*Indicate your interest in the resolution of this issue by adding the 👍 reaction. Comments such as "+1" will be removed.*

### PR DESCRIPTION
**Pull request by gugu**
## Pull Request Checklist

- [x] Have you added new tests to prevent regressions?
- [x] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [x] Did you update the typescript typings accordingly (if applicable)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

Closes #15053 

To prevent string concatenation and keep addition correct I've added a cast to BigInt
To keep existing API result contains either number or string instead of bigint

### CODE REVIEW

**Comment by gugu** [comment]
@ephys can I write unit test for this `formatData` instead? I think it will be much simpler and faster

**Comment by gugu** [comment]
> Type of value
Sequelize returns strings for BigInts larger than Number.MAX_SAFE_INTEGER, so I tried to be consistent with that behaviour

**Comment by ephys** [comment]
Unfortunately `formatResults` is an internal function specific to mysql, whereas an integration test would be able to run on all dialects (we won't ask you to fix the other dialects if they fail too). An integration test would also be more resilient to the upcoming rewrite of all query.js files

**Comment by ephys** [comment]
> Sequelize returns strings for BigInts larger than Number.MAX_SAFE_INTEGER

That is true (depending on the dialect). There are a lot of inconsistencies with DataTypes but https://github.com/sequelize/sequelize/pull/14505 will finally fix all of that

**Comment by gugu** [comment]
I wrote tests and changed types to match input. The test fails for SQLite, but I think it is a part of a larger problem, which is out of scope of the fix

**Comment by gugu** [comment]
@ephys is there anything extra I need to change in this PR?

File: test/integration/model.test.js [hunk 1]
```diff
@@ -2843,5 +2843,29 @@ describe(Support.getTestDialectTeaser('Model'), () => {
       const users1 = await user.findAll();
       expect(users1[0].username).to.equal('jon');
     });
+
+    it('should correctly set identifiers in a column with autoIncrement with bigint values', async function () {
+      if (dialectName === 'sqlite') {
+        return;
+      }
+
+      const user = this.sequelize.define('User', {
```
Comment (ephys) [inline test/integration/model.test.js:2852]: To follow our conventions (it's not respected in old code but we're slowly refactoring everything)

```suggestion
      const User = this.sequelize.define('User', {
```

File: test/integration/model.test.js [hunk 2]
```diff
@@ -2843,5 +2843,29 @@ describe(Support.getTestDialectTeaser('Model'), () => {
       const users1 = await user.findAll();
       expect(users1[0].username).to.equal('jon');
     });
+
+    it('should correctly set identifiers in a column with autoIncrement with bigint values', async function () {
+      if (dialectName === 'sqlite') {
```
Comment (ephys) [inline test/integration/model.test.js:2848]:
```suggestion
      // sqlite returns bigints as numbers https://github.com/sequelize/sequelize/issues/11400
      if (dialectName === 'sqlite') {
```

**Review by ephys — CHANGES_REQUESTED** [review]
This looks good to me, here are two minor changes in the test :)

Sorry for the delay, it's difficult to find the time to review during the work week

**Review by gugu — COMMENTED** [review]
capitalize class name in test

**Review by ephys — APPROVED** [review]
Thanks :)

**Pull request merged**